feat: added Python API

camilovelezr opened the pull request:
This PR introduces a new Python API for WIC.

Purpose of the API: to provide programmatic access for building workflows.

Usage:
Assume the user wants to create a two step workflow. The user already has one CLT for each step.
```python3
from pathlib import Path
from wic import Step, Workflow
step1 = Step("/home/user/clt1.cwl")
step2 = Step("/home/user/clt2.cwl")

step1.inpDir = Path("/home/user/input")
step1.darkField = True
step2.inpDir = step1.outDir # explicit link

workflow = Workflow([step1, step2], "two_step_workflow", path=Path("/home/user/two_step_workflow"))
workflow.compile()
workflow.run() # optional -> will use cwltool
```
`Workflow.compile()` will:
1. Generate the relevant WIC YAML - defining which tools, in what order, with what parameters - and dump it to disk.
2. Call WIC from the command line to compile the YAML that was just dumped to disk

`Workflow.run()` will run the workflow generated by WIC, specifying the inputs YML generated by WIC, and using `cwltool` with `--relax-path-checks`

hamshkhawar commented on src/wic/api/_types.py:
```diff
@@ -0,0 +1,30 @@
+    "double": float,
+    "string": str,
+    # "File": CWLFile,
+    # "Directory": CWLDir,
```
Could you please remove duplicated keys and values from CWL_TYPES_DICT dictionary?

camilovelezr replied:
Addressed in #135

hamshkhawar commented on src/wic/api/pythonapi.py:
```diff
@@ -0,0 +1,364 @@
+            text=True,
+            universal_newlines=True,
+        )
+        print(proc.stdout)
```
Replace "print" with by using logger.info() as you already using logging package

camilovelezr replied:
Addressed in #135

hamshkhawar reviewed (commented):
@camilovelezr I have added couple of comments on this PR. I haven't tested this code? How can I do it?

camilovelezr commented:
> @camilovelezr I have added couple of comments on this PR. I haven't tested this code? How can I do it?

@hamshkhawar Thank you Hamdah, I addressed your comments and responded. I will soon add unit tests
